### PR TITLE
[Caching] SwiftDriver Caching API refinement for build system

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Add/adjust APIs from swift-driver for build system integration.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
This relies on https://github.com/apple/swift/pull/69934 to be functional.

**Comment by cachemeifyoucan** [comment]
ping

**Comment by cachemeifyoucan** [comment]
ping

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

File: Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift [hunk 1]
```diff
@@ -37,6 +37,36 @@ public final class CachedCompilation {
     lib.api.swiftscan_cached_compilation_is_uncacheable(ptr)
   }
 
+  public func makeGlobal(_ callback: @escaping (Swift.Error?) -> ()) {
```
Comment (akyrtzi) [inline Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift:40]: There's duplication of code between this and `makeGlobal() async`, I'd suggest to re-write `makeGlobal() async` in terms of calling `makeGlobal(_ callback:)`.

**Review by akyrtzi — APPROVED** [review]
One suggested change, otherwise LGTM.

File: Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift (hunk 1 above)
Comment (cachemeifyoucan) [reply Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift:40]: Good catch. I also forgot the `async` version of `makeGlobal` can just return `Void` as error is indicated via throw.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Pull request merged**